### PR TITLE
Add creality ramps pin definition

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Add pins file for Creality ramps generation boards with definition for filament sensor and heater 1 pins. Update creality melzi to support servo0 definition for bltouch when required. Standard method using TH3D EZOut boards, provided as stock upgrade by Tinymachines3D, and by many users.

### CODE REVIEW

**Comment by thinkyhead** [comment]
- Do you think `PIN_EXP1` - `PIN_EXP4` should also be moved to the `pins_RAMPS_CREALITY.h` file?
- Does this only apply to the CR-10, and not any other Creality3D models?

**Comment by InsanityAutomation** [comment]
They're spare pins not used for anything on a header on the pcb. None of the creality models utilize them at the moment. They very well could be, I've been using d12 as a case light and the analog to control chamber temp sensor and fan.

**Comment by thinkyhead** [comment]
- Does `MOTHERBOARD BOARD_RAMPS_CREALITY` only apply to the CR-10, or should it also apply to some other Creality3D models? The "Ender 4" was mentioned in the pins file until I changed it.

**Comment by InsanityAutomation** [comment]
In the boards.h file, I listed the other machines that applies to. Basically just CR 20 and CRX.

File: Marlin/src/pins/pins_RAMPS_CREALITY.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#endif
+
+#define BOARD_NAME "Creality3D RAMPS"
+
+//
+// Heaters / Fans
+//
+
+// Power outputs EFBF or EFBE
+#define MOSFET_D_PIN 7
```
Comment (thinkyhead) [inline Marlin/src/pins/pins_RAMPS_CREALITY.h:34]: @InsanityAutomation — For this board, should it be 10:E0, 9:E1, 8:Bed, 7:Fan or 10:E0, 7:E1, 8:Bed, 9:Fan ?? See 7b25b9ec5b.

**Pull request merged**